subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/DefaultJavaToolchainRepositoryHandler.java

```diff
@@ -92,12 +103,97 @@ public void repository(String name, Action<? super JavaToolchainRepository> conf
     }
 
     @Override
-    public List<JavaToolchainRepository> repositories() {
-        return Collections.unmodifiableList(repositories);
+    public List<JavaToolchainRepository> getAsList() {
+        ArrayList<JavaToolchainRepository> copy = repositories.stream()
+                .map(ImmutableJavaToolchainRepository::new)
+                .collect(Collectors.toCollection(ArrayList::new));
+        return Collections.unmodifiableList(copy);
+    }
+
+    public List<JavaToolchainRepository> getAsLiveList() {
```

Comment:
This is more than just a live list. Its a whole mutable reference which completely side-steps `disableFurtherMutations` -- IMO it should be named as such. Perhaps `getMutableRepositories` or `getMutableList`?

Reply:
Yeah, I don't like it either. We'll take a second look at it.

Reply:
From what I can see, the only usage made of this method does not even require it. It could use `getAsList()`

Reply:
It could use `getAsList` -- though I imagined we were using the "live list" as an optimization so we don't need to wrap all of the repositories in the immutable wrapper if we're just going to immediately pull out the implementation class. 

Though, I imagine the performance implications are negligible and using `getAsList` should be fine

Reply:
Removing the live view. As it was `getAsList()` could not replace it, but made it so that it can now.